File: DSA/binarySearch.py
l = [1,2,4,7,9,10,14,16,18,30,60,80,82,84,86,91,97,102]



def binarySearchInt(array,left,right,value):
			
	while (left <= right):
		mid = (left + right) //2
		
		if (array[mid] == value):
			return mid
		
		elif(array[mid] > value):
			right = mid-1
		else:
			left = mid+1
	return -1
		 


def binarySearchRec(array,left,right,value):
	
	mid = (left + right)//2
	
	if (left > right):
		return -1
	
	elif(array[mid] == value):
		return mid
	
	elif(array[mid] > value):
		return binarySearchRec(array,left,mid-1,value)
	
	else:
		return binarySearchRec(array,mid + 1, right,value)
		


print(binarySearchRec(l,0,len(l)-1,9))
print(binarySearchInt(l,0,len(l)-1,9))
	
		

	

File: Java/BinarySearch.java
import java.util.Scanner;

public class BinarySearch {

    // Binary search function
    public static int binarySearch(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            // Check if the target is at the middle
            if (arr[mid] == target) {
                return mid;
            }

            // If the target is greater, ignore the left half
            if (arr[mid] < target) {
                left = mid + 1;
            }
            // If the target is smaller, ignore the right half
            else {
                right = mid - 1;
            }
        }

        // Target not found in the array
        return -1;
    }

public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Prompt the user to enter the array using scanner class
        System.out.print("Enter the sorted array (comma-separated values): "); //input type comma-separated values
        String input = scanner.nextLine();
        String[] elements = input.split(",");
        int[] arr = new int[elements.length];

        for (int i = 0; i < elements.length; i++) {
            arr[i] = Integer.parseInt(elements[i].trim());
        }

        // Prompt the user to enter the target element using scanner class
        System.out.print("Enter the target element: ");
        int target = scanner.nextInt();

        int result = binarySearch(arr, target);

        if (result != -1) {
            System.out.println("Target found at index " + result);
        } else {
            System.out.println("Target not found in the array.");
        }

        scanner.close();
    }
}
